### PR TITLE
Fix rel delete and create

### PR DESCRIPTION
**Pull request by ray6080**

### CODE REVIEW

File: src/storage/store/column_chunk.cpp [hunk 1]
```diff
@@ -473,6 +473,29 @@ class FixedListColumnChunk : public ColumnChunk {
     }
 };
 
+class InternalIDColumnChunk final : public ColumnChunk {
```
Comment (acquamarin) [inline src/storage/store/column_chunk.cpp:476]: Should we move the class to .h ?

File: src/storage/store/rel_table_data.cpp [hunk 2]
```diff
@@ -291,6 +308,160 @@ void RelTableData::prepareCommitCSRNGWithoutSliding(Transaction* transaction,
...
+        column->getDataType()->copy(), enableCompression, oldCapacity);
+    column->scan(transaction, nodeGroupIdx, columnChunk.get());
+    auto newColumnChunk = ColumnChunkFactory::createColumnChunk(
+        column->getDataType()->copy(), enableCompression, newCapacity);
+    auto currentNumSrcNodesInNG = csrOffsetChunk->getNumValues();
+    auto csrOffsets = (offset_t*)csrOffsetChunk->getData();
+    auto relIDs = (offset_t*)relIDChunk->getData();
+    auto nodeGroupStartOffset = StorageUtils::getStartOffsetOfNodeGroup(nodeGroupIdx);
+    for (auto i = 0; i < currentNumSrcNodesInNG; i++) {
+        auto nodeOffset = nodeGroupStartOffset + i;
```
Comment (acquamarin) [inline src/storage/store/rel_table_data.cpp:409]: You don't have to slide node one by one if that node doesn't have any updates.
You can accumulate nodes, until you encounter a node that has updates

File: src/storage/store/rel_table_data.cpp [hunk 3]
```diff
@@ -291,6 +308,160 @@ void RelTableData::prepareCommitCSRNGWithoutSliding(Transaction* transaction,
...
+        auto hasDeletions = deleteInfo.contains(nodeOffset);
+        auto hasUpdates = updateInfo.contains(nodeOffset);
+        auto hasInsertions = insertInfo.contains(nodeOffset);
+        if (!hasDeletions && !hasUpdates && !hasInsertions) {
+            // Append the whole csr.
+            newColumnChunk->append(
+                columnChunk.get(), startCSROffset, endCSROffset - startCSROffset);
+            continue;
+        }
+        for (auto csrOffset = startCSROffset; csrOffset < endCSROffset; csrOffset++) {
```
Comment (acquamarin) [inline src/storage/store/rel_table_data.cpp:421]: These sliding functions are too long, better to create some helpers

**Review by acquamarin — APPROVED** [review]

File: src/storage/store/column_chunk.cpp (hunk 1 above)
Comment (ray6080) [reply src/storage/store/column_chunk.cpp:476]: It's actually not exposed and used somewhere, so I think better kept inside cpp file to reduce header file size.

File: src/storage/store/rel_table_data.cpp (hunk 3 above)
Comment (ray6080) [reply src/storage/store/rel_table_data.cpp:421]: Yep. In the progress for that now.

File: src/storage/store/rel_table_data.cpp (hunk 2 above)
Comment (ray6080) [reply src/storage/store/rel_table_data.cpp:409]: Separate this into another PR.

File: src/storage/store/rel_table_data.cpp (hunk 3 above)
Comment (ray6080) [reply src/storage/store/rel_table_data.cpp:421]: Separate this into another PR.

**Pull request merged**